backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectRepository.java

```diff
@@ -289,13 +290,20 @@ public int getCountByProjectId(String id) {
 
     @NotNull
     private Set<Project> filterAccessibleProjectsByIds(User user, Set<String> searchIds) {
-        final Set<Project> accessibleProjects = getAccessibleProjects(user);
-
         final Set<Project> output = new HashSet<>();
-        for (Project accessibleProject : accessibleProjects) {
-            if (searchIds.contains(accessibleProject.getId())) output.add(accessibleProject);
-        }
 
+        searchIds.stream().filter(CommonUtils::isNotNullEmptyOrWhitespace).forEach(searchId -> {
```

Comment:
I hope, this is not related to `moderation request`?

Reply:
`filterAccessibleProjectsByIds(User user, Set<String> searchIds)` is getting called when a moderation request is opened.
Called from ModerationPortlet.class -> `renderProjectModeration()` ->`prepareProject()`.
Similarly called from ModerationPortlet.class -> `renderComponentModeration()` ->`prepareComponent()`.
Similarly called from ModerationPortlet.class -> `renderReleaseModeration()` ->`prepareRelease(`).

Therefore optimized code here, since it was loading all projects before.

Reply:
Missed it. Thanks for the explanation.